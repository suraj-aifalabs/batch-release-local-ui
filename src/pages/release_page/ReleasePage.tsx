import { Button } from '@/components/ui/button';
import React, { useState } from 'react'
import { FillAndPreviewPDF } from './ViewPDF';

const ReleasePage = () => {
    const [showPdfViewer, setShowPdfViewer] = useState(false);
    const [view, setView] = useState(true)
    const qaData = {
        "_id": "670cb671eb96e6534cd889da",
        "stage": "ATLAS",
        "status": "Ready for Sign",
        "cequenceId_Hashed": "2119fdc4d0c0444318bfc5a7e8e2ac7c:351e4e7960f9a3a377a0e28e9ae103e5a4aa47f73cb2f7c1ff4f790409444edd9a7fab1b23a029faf789b11f115fa3ea",
        "patientName": "Sanjay",
        "patientDOB": "2024-10-14T06:13:05.105Z",
        "patientWeight": "78",
        "cquenceDIN": "DIN00777",
        "cquenceOrderId": "ORD15",
        "country": "IN",
        "coicBagId": "COIC883",
        "batchNumber": "B114AF",
        "createdAt": "2024-10-14T06:13:05.105Z",
        "updatedAt": "2024-10-14T06:14:55.512Z",
        "__v": 0,
        "pccNumber": "PCC8787",
        "totalVolume": "100",
        "productDose": "10mg",
        "marketAuthorizationNumber": "123564",
        "nameAndAddress": "David Laid Seattle WA 90878",
        "productNDC": "123",
        "expirationDate": "2024-10-14T06:13:05.105Z"
    }



    const handleViewPDF = () => {
        console.log("in handle view");
        setView(false)
        setShowPdfViewer(true);
    };
    return (
        <div>
            <div style={{ display: 'flex', flexDirection: 'column', gap: '20px', justifyContent: 'space-between' }}>
                {view &&
                    <Button onClick={handleViewPDF} style={{ width: '200px', display: 'flex', alignSelf: 'center', marginTop: '10px', justifyContent: 'center' }}>
                        View PDF
                    </Button>
                }
                {
                    showPdfViewer && (
                        <div style={{ display: 'flex', alignSelf: 'center', marginTop: '20px', justifyContent: 'center' }}>
                            <FillAndPreviewPDF

                                data={qaData}

                            />
                        </div>
                    )
                }

            </div >
        </div>
    )
}

export default ReleasePage
