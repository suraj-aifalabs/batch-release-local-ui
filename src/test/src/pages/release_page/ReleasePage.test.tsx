import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReleasePage from '../../../../pages/release_page/ReleasePage';
import ViewPdf from '../../../../pages/release_page//ViewPdf';

jest.mock('../../../../pages/release_page//ViewPdf', () => ({
    __esModule: true,
    default: jest.fn(() => <div data-testid="fill-preview-mock">Mock FillAndPreviewPDF</div>)
}));


describe('ReleasePage component', () => {
    it('renders without crashing', () => {
        render(<ReleasePage />);
        expect(screen.getByTestId('fill-preview-mock')).toBeInTheDocument();
    });

    it('renders the FillAndPreviewPDF component', () => {
        render(<ReleasePage />);
        expect(ViewPdf).toHaveBeenCalled();
    });

    it('has correct layout structure', () => {
        const { container } = render(<ReleasePage />);

        const outerDiv = container.firstChild;
        expect(outerDiv).toBeInTheDocument();
        expect(outerDiv).toHaveStyle('display: block'); // Default div display

        const flexContainer = outerDiv?.firstChild;
        expect(flexContainer).toHaveStyle({
            display: 'flex',
            flexDirection: 'column',
            gap: '20px',
            justifyContent: 'space-between'
        });

        const nestedFlexContainer = flexContainer?.firstChild;
        expect(nestedFlexContainer).toHaveStyle({
            display: 'flex',
            alignSelf: 'center',
            marginTop: '20px',
            justifyContent: 'center'
        });
    });

    it('positions FillAndPreviewPDF correctly', () => {
        render(<ReleasePage />);
        const fillPreviewComponent = screen.getByTestId('fill-preview-mock');
        const parentDiv = fillPreviewComponent.parentElement;

        expect(parentDiv).toHaveStyle({
            display: 'flex',
            alignSelf: 'center',
            marginTop: '20px',
            justifyContent: 'center'
        });
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<ReleasePage />);
        expect(asFragment()).toMatchSnapshot();
    });
});