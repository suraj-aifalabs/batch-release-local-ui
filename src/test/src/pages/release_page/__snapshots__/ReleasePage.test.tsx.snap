// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleasePage component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="display: flex; flex-direction: column; gap: 20px; justify-content: space-between;"
    >
      <div
        style="display: flex; align-self: center; margin-top: 20px; justify-content: center;"
      >
        <div
          data-testid="fill-preview-mock"
        >
          Mock FillAndPreviewPDF
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
